### DISCUSSION TITLE
Regarding the transfer mechanism for Large SPDM Message (CHUNK_SEND and CHUNK_GET)

### DISCUSSION THREAD

**Discussion by KwonHyunSoo** [DMTF/libspdm#1012] [Q&A]
Hi,

According to _10.26.3 Additional chunk transfer requirements_ of the SPDM v1.2 version 1.2.0 (or 1.2.1), due to retries, an SPDM endpoint might re-send or re-retrieve the current chunk number in the transfer.
However, it seems that libspdm does not consider such re-send mechanism in that the responder returns the error response when receiving the sequence number in SPDM request is not equal to "send_info->chunk_seq_no + 1" as follows: 
https://github.com/DMTF/libspdm/blob/09be8c3d9bc18f23ba8e2fc7fa3728551836570c/library/spdm_responder_lib/libspdm_rsp_chunk_send_ack.c#L147-L153

How can the responder check whether the current CHUNK_SEND command is re-transmitted or not?

Thanks, 
H. Kwon

**Comment by jyao1** [comment]
libspdm does not support retry so far.
ref: https://github.com/DMTF/libspdm/issues/491

**Comment by steven-bellock** [comment]
Retries with chunking are a bit more straightforward than other cases since chunking includes the sequence number. Also the SPDM specification does not require that a Responder support retries during chunking, just that 
>When transferring a large SPDM message, an SPDM endpoint shall be prohibited from transferring a chunk sequence number (that is, ChunkSeqNo) less than the current chunk sequence number.
